bindings/java/src/java_module_host.c

```diff
@@ -239,8 +248,41 @@ static MODULE_HANDLE JavaModuleHost_Create(BROKER_HANDLE broker, const void* con
                                                 }
                                                 else
                                                 {
-                                                    jobject jModule_object = NewObjectInternal(result->env, jModule_class, jModule_constructor, 3, (jlong)result, jBroker_object, jModule_configuration);
-                                                    exception = JNIFunc(result->env, ExceptionOccurred);
+                                                    jobject jModule_object;
+                                                    if (emptyConstructor)
```

Comment:
Suggestion: some descriptive comments as to what's going on here might help people who aren't too familiar with JNI.